client/src/app/pages/applications/components/application-detail-drawer/components/assessed-archetypes.tsx

```diff
@@ -0,0 +1,51 @@
+import React from "react";
+import { Ref } from "@app/api/models";
+import { Label, LabelGroup, Spinner } from "@patternfly/react-core";
+import { EmptyTextMessage } from "@app/components/EmptyTextMessage";
+import { useTranslation } from "react-i18next";
+import { useFetchArchetypes } from "@app/queries/archetypes";
+import { useFetchAllAssessmentsWithArchetypes } from "@app/queries/assessments";
+
+interface IAssessedArchetypesProps {
+  archetypeRefs: Ref[] | undefined;
+}
+
+export const AssessedArchetypes: React.FC<IAssessedArchetypesProps> = ({
+  archetypeRefs,
+}) => {
+  const { t } = useTranslation();
+  const { archetypes, isFetching: isFetchingArchetypes } = useFetchArchetypes();
+  const applicationArchetypes = archetypes.filter(
+    (archetype) => archetypeRefs?.some((ref) => ref.id === archetype.id)
+  );
```

Comment:
Just a thought -- this kind of thing could be embedded directly in the hook so the hook use could look like:
```js
  const { archetypes, ... } = useFetchArchetypes({ 
    includeIds: archetypeRefs.map((a) => a.id) 
  });
```

or even somethings more like:
```js
  const { archetypes, ... } = useFetchArchetypes({
    forApplication: application
  });
```

Then if every the rest endpoint gets a filter itself, that can rework can be handled directly in the query hook.

Reply:
Interesting - I wonder if the api supports this?